src/vivarium_nih_us_cvd/components/treatment.py

```diff
@@ -719,15 +719,23 @@ def apply_sbp_treatment_ramp(
             pop_visitors[data_values.COLUMNS.SBP_MEDICATION]
             != self.sbp_treatment_map[max(self.sbp_treatment_map)]
         ].index
-        medication_change = to_prescribe_b.union(
-            to_prescribe_d.intersection(adherent).intersection(not_already_max_medicated)
+        changing_prescribed_medication = to_prescribe_d.intersection(adherent).intersection(
+            not_already_max_medicated
         )
+        medication_change = to_prescribe_b.union(changing_prescribed_medication)
         pop_visitors.loc[medication_change, data_values.COLUMNS.SBP_MEDICATION] = (
             pop_visitors[data_values.COLUMNS.SBP_MEDICATION].map(
                 {v: k for k, v in self.sbp_treatment_map.items()}
             )
             + 1
         ).map(self.sbp_treatment_map)
+        # update therapeutic inertia for people on medication going up a level
+        self.dynamic_therapeutic_inertia_propensity.loc[
+            changing_prescribed_medication
+        ] = self.randomness.get_draw(
+            changing_prescribed_medication,
+            additional_key="dynamic_therapeutic_inertia_" + str(self.clock()),
```

Comment:
You don't need to include the clock in the additional seed. This is automatically part of the randomness stream